trapping-rainwater.md

```diff
@@ -0,0 +1,21 @@
+ Given n non-negative integers representing an elevation map where the width of each bar is 1, compute how much water it is able to trap after raining.
+
+### Example ###
+Given [0, 1, 0, 2, 1, 0, 1, 3, 2, 1, 2, 1] as evaluation map:
+
+                                +---+
+                +---+           |   |---+   +---+
+        +---+   |   |---+   +---|   |   |---|   |---+
+        |   |   |   |   |   |   |   |   |   |   |   |
+    -------------------------------------------------
+    | 0 | 1 | 0 | 2 | 1 | 0 | 1 | 3 | 2 | 1 | 2 | 1 |
+after the rain it was so:
+
+                                +---+
+                +---+$$$$$$$$$$$|   |---+$$$+---+
+        +---+   |   |---+   +---|   |   |---|   |---+
+        |   |$$$|   |   |$$$|   |   |   |   |   |   |
+    -------------------------------------------------
```

Comment:
Why is 2 only 1.5 times as high as 1 is, and 3 is twice as high as 1?
Also why is there a row without water in the middle lake?

The current graphs are somewhat confusing and should get repaired to make it more clear.